### PR TITLE
simplified a2m

### PR DESCRIPTION
**Pull request by themighty1**
This PR simplifies a2m
It is easier to reason about the protocol if we break it down in 2 steps: (sender's additive share is x, receiver's is y)
1. the sender sends to peer `y * random`
2. the sender mixes in `x * random` into the last OT.

### CODE REVIEW

**Comment by themighty1** [comment]
I agree with your point about diverging from the paper. I lost sight of that.

To stay closer to the paper, I suggest, instead of adding x_A to the last OT, we can do what the modexp paper says in Fig 7 :
Screenshot from 2023-02-27 12-27-17

We should decompose `x_A` into 128 *random* components and mix `random * t_i` into each OT that we send. Note that as it is implemented now, we decompose `x_A` bit-by-bit, but in order to stay closer to the paper, we should decompose it randomly.
What do you think?

**Review by th4s — APPROVED** [review]
:rocket: ack

**Comment by themighty1** [comment]
@sinui0 , you can review now

File: mpc/share-conversion-core/src/a2m.rs [hunk 1]
```diff
@@ -26,46 +26,58 @@ impl<T: Field> AddShare<T> {
...
+        // split up our additive share `x` into random summands
+        let mut x_summands: Vec<T> = vec![T::zero(); T::BIT_SIZE as usize];
+        x_summands.iter_mut().for_each(|s| *s = T::rand(rng));
+
+        // set the last summand such that the sum of all [T::BIT_SIZE] summands equals `x`
+        x_summands[T::BIT_SIZE as usize - 1] = self.inner()
+            + -x_summands
+                .iter()
+                .take(T::BIT_SIZE as usize - 1)
+                .fold(T::zero(), |acc, i| acc + *i);
```
[the remark this answers is not in the document]
Comment (themighty1) [reply mpc/share-conversion-core/src/a2m.rs:49]: When I tried:
`x_summands[T::BIT_SIZE as usize - 1] =
            self.inner() + (-x_summands[..T::BIT_SIZE as usize - 1].iter().sum::<T>());`

I got:

`the trait `Sum<&T>` is not implemented for `T``

are you ok with proceding as it is rn without implementing the Sum trait?

**Pull request merged**